### DISCUSSION TITLE
Side by Side Grip

### DISCUSSION THREAD

**Discussion by TheOmniJuggler** [jkboyce/jugglinglab#128] [Q&A]
Hey awesome jugglers!
Is it possible in Juggling Lab to adjust the position of the balls when they are in the hand? Specifically for multiplex patterns where a side by side grip is essential to conveying the pattern (even split multiplexes, claymotion patterns, etc.)
I'm fairly familiar with JL and haven't been able to find a way to do it, but I figured I'd ask in case I missed it. 
If it's not possible, is it something that could ever be implemented? Or does it clash with the way JL works? 

Thanks for reading!
![EvenSplitMX](https://user-images.githubusercontent.com/92499991/226756351-861b6aca-a905-4f04-9082-91d9a5c4ae36.jpg)
_attached an image of what I'm describing_

**Comment by jkboyce** [comment]
Hi @TheOmniJuggler , this is an excellent question!

The simple answer is no, Juggling Lab can't accurately render multiple balls in the hand like this. It always draws balls in the hand at the same location, making them overlap.

This isn't exactly true. If you really wanted to generate this effect you could do so, in a very cumbersome way: You could have three jugglers at the same location in space, with two of them hidden (there is a `hidejugglers` option in the animation preferences). Hidden juggler 1 would hold the blue ball in your drawing, hidden juggler 2 would hold the green ball, and visible juggler 3 would hold an invisible ball ("transparent" is a ball color). Then you set the throw/catch locations for all three jugglers to get the ball positions you want, both in the hand and in the air. Certainly not a convenient thing to set up!

To add this feature there would be some details to work out, in particular what happens when you catch multiple balls into the hand in sequence. Does the first catch go into the center of the hand? If so, how does it then move over when the second catch comes in? Or does the first catch go into its slot to one side of the center? I'm confident it would be possible to sort out something that would render well, but nobody's ever taken the time. This would be a good project if you're interested in programming!

**Reply by TheOmniJuggler** [reply]
Thanks for getting back to me! I'm definitely going to look into the multiple jugglers approach. I have never played with animating more than one juggler before in JL, so this should be a fun exploration for me!
As for the programming, sadly my skills are so pathetically amateur I can barely understand the code that exists in the JL files as they exist right now, let alone trying to realistically add my own :) But I am trying to learn slowly, so maybe someday I'll be able to do it!

A few questions, just for my fun attempt at understanding how JL works. Sorry if they're super naive or noob:

I'm less interested in allowing the animator to create side-by-side multiplexes with a quick animation preset or anything like that. More interested in whether it could allow the user to manually achieve this in the visual or pattern editors by dragging the positions of each keyframe. Could the issue you mentioned of JL not knowing which spot to default to for each catch be simplified by requiring the user to keyframe hand/ball positions if they wanted that effect? Or do those have to go hand in hand? (pun intended hehe)
It seems like the visual editor does understand there are 3 separate keyframe-able objects of a multiplex catch/hold/throw on the timeline; 1 for each ball and 1 for the hand.  Does JL actually see these three points as separate? Or do they somehow get interpreted as a combined point once they reach the hand? (I hope that question makes sense...I'm so new to this) 
I think what I'm actually trying to ask is, with the way JL currently works, does it theoretically have 3 different keyframes that are just coded so they are tied together in a hold, which means a change in code could potentially allow those 3 keyframes (objects, whatever you call them) to be moved in space separately? Or is the fundamental coding of JL done in a way that those 3 keyframes are not seen as independent at all in this instance, so adding this feature would require changing how JL sees catches and holds all together? 

Again, super sorry if any of those questions are frustrating or too hard to answer with my limited JL understanding. I suspect it might feel similar to when one tries to explain that siteswap is actually about timing of throws rather than height to a new juggler with limited understand of how juggling works at all. If that's the case, feel free to just say that it's probably a bit over my head :) I'll totally understand. Either way, thanks for the answer you already gave. I'm looking forward to learning more about JL works!
